### ISSUE TITLE
GRUB sparse file error on minimal image

### ISSUE THREAD

**Issue by LukeShortCloud** [LukeShortCloud/winesapOS#451]
The GRUB sparse file error is back but now on our minimal image. It does not prevent the system from booting but it can be a concerning false-positive message to users.

Related: https://github.com/LukeShortCloud/winesapOS/issues/27

**Comment by LukeShortCloud** [comment]
Here's the error but not key press was required. It would automatically continue to boot after a couple of seconds.

```
error: sparse file not allowed.
Loading Linux linux-lts ...
Loading initial ramdisk ...

Press any key to continue...
```

**Comment by LukeShortCloud** [comment]
This suggested workaround does NOT work:

```
sudo grub-install --recheck --no-rs-codes --modules="btrfs zstd part_gpt part_msdos"
sudo grub-mkconfig -o /boot/grub/grub.cfg
```

I also tried:

```
sudo grub-install --recheck --no-rs-codes --modules="btrfs zstd part_gpt part_msdos" --target=x86_64-efi --efi-directory=/boot/efi --bootloader-id=winesapOS --removable
sudo grub-mkconfig -o /boot/grub/grub.cfg
```

It is important to note that we do need support for all of those modules in winesapOS: btrfs, zstd, GPT, and MSDOS (MBR).

https://forum.manjaro.org/t/grub-error-sparse-file-not-allowed/20267/2

**Comment by LukeShortCloud** [comment]
This is actually not an issue specifically with the minimal image. It was ever since we allowed changing the default Linux kernel which is a huge feature we want to keep.

**Comment by LukeShortCloud** [comment]
Apparently this error is because GRUB does not support Btrfs. We currently have `/boot` on the root `/` partition which is Btrfs. Someone suggested re-using the FAT32 EFI partition for GRUB. Alternatively, we could have a separate partition for `/boot` formatted as ext4.

https://www.reddit.com/r/archlinux/comments/mqktka/comment/gugv3qw/

**Comment by LukeShortCloud** [comment]
I built a secure image and it does not have the problem. I also noticed that `/boot` is mounted in the secure image but was not in the `minimal`. I found that there was some missing logic to account for the exFAT storage partition not existing on the minimal image. This lead to `/boot` not being mounted, which lead to it not being in `/etc/fstab`/, which means that GRUB is installed onto the Btrfs root file system instead of the ext4 boot partition. I am testing out a fix to this now.
